imap/httpd.c

```diff
@@ -2392,12 +2392,15 @@ static int parse_connection(struct transaction_t *txn)
                 if (!strcasecmp(token, "upgrade")) {
                     /* Client wants to upgrade */
                     parse_upgrade(txn);
+                    break;
                 }
                 else if (!strcasecmp(token, "close")) {
                     /* Non-persistent connection */
                     txn->flags.conn |= CONN_CLOSE;
+                    break;
                 }
-                break;
+
+                GCC_FALLTHROUGH
```

Comment:
If we look at the line a few below this (I can't tag it in the comment, because it's outside the context, doh) we see:

```
                    /* Persistent connection */
                    txn->flags.conn = CONN_KEEPALIVE;
```

Does that need to be changed to:

```
                    txn->flags.conn |= CONN_KEEPALIVE;
```

Since http/1.1 allows additional connection flags.

Reply:
This logic change was incorrect and has been fixed in subsequent commit